File: test/bot_sources/kotlin/basic_bot.kt
fun main(args : Array<String>) {
    val actions = listOf("rock", "paper", "scissors")

    while (true) {
        val line = readLine()
        if (line == null) {
            break
        }

        val stripped = line.trim()
        if (stripped.length == 0) {
            continue
        }

        val words = line.split(" ");
        if (words.size == 0) {
            continue
        }

        when (words.get(0)) {
            "action" -> {
                val choice = (Math.random() * 3).toInt()
                println(actions.get(choice))
            }
        }
    }
}

File: test/bot_sources/cs/basic_bot.cs
using System;
using System.Collections;
using System.IO;

class Bot {
    static void Main() {
        string line;

        while(true) {
            line = Console.ReadLine();
            if (line == null) {
                break;
            }

            string[] words = line.Split(' ');
            if (words.Length == 0) {
                continue;
            }

            if (words[0] != "action") {
                continue;
            }
            
            Random rnd = new Random();
            int choice = rnd.Next(1, 4);
            
            if (choice == 1) {
                Console.Write("rock\n");
            }
            else if (choice == 2) {
                Console.Write("paper\n");
            }
            else if (choice == 3) {
                Console.Write("scissors\n");
            }
        }
    }
}
